include/ignition/sensors/SensorFactory.hh

```diff
@@ -41,32 +40,35 @@ namespace ignition
     class SensorFactoryPrivate;
 
     /// \brief Base sensor plugin interface
+    /// \deprecated Sensor plugins are deprecated. Instantiate sensor objects
+    /// instead.
     class IGNITION_SENSORS_VISIBLE SensorPlugin
     {
-      /// \brief Allows using shorter APIS in common::PluginLoader
-      public: IGN_COMMON_SPECIALIZE_INTERFACE(ignition::sensors::SensorPlugin)
-
       /// \brief Instantiate new sensor
       /// \return New sensor
-      public: virtual Sensor *New() = 0;
+      public: virtual Sensor * IGN_DEPRECATED(6) New() = 0;
```

Comment:
Windows is failing in this line https://build.osrfoundation.org/job/ign_sensors-pr-win/676/consoleFull#-34390258ff58640-3599-4406-a210-216932f1748c

I remember that @adlarkin mentioned something about `IGN_DEPRECATED` failures when the methods are returning raw pointers